examples/nucleo_g474re/ixm42xxx_fifo/main.cpp

```diff
@@ -0,0 +1,154 @@
+        }
+
+        MODM_LOG_INFO << "IXM-42xxx Initialized" << modm::endl;
+        MODM_LOG_INFO << "Fifo Buffer Size: " << fifoData.getFifoBufferSize() << modm::endl;
+
+        /// Configure FIFO
+        PT_CALL(imu.updateRegister(Register::FIFO_CONFIG, FifoMode_t(FifoMode::StopOnFull)));
+        PT_CALL(imu.updateRegister(Register::FIFO_CONFIG1, FifoConfig1::FIFO_TEMP_EN | FifoConfig1::FIFO_GYRO_EN | FifoConfig1::FIFO_ACCEL_EN));
+        PT_CALL(imu.writeFifoWatermark(1024));
+
+        /// Configure interrupt
+        PT_CALL(imu.updateRegister(Register::INT_CONFIG, IntConfig::INT1_MODE | IntConfig::INT1_DRIVE_CIRCUIT | IntConfig::INT1_POLARITY));
+        PT_CALL(imu.updateRegister(Register::INT_CONFIG1, IntConfig1::INT_ASYNC_RESET));
+        PT_CALL(imu.updateRegister(Register::INT_SOURCE0, IntSource0::FIFO_THS_INT1_EN | IntSource0::FIFO_FULL_INT1_EN, IntSource0::UI_DRDY_INT1_EN));
+
+        /// Configure data sensors
+        PT_CALL(imu.updateRegister(Register::GYRO_CONFIG0, GyroFs_t(GyroFs::dps2000) | GyroOdr_t(GyroOdr::kHz1)));
+        PT_CALL(imu.updateRegister(Register::ACCEL_CONFIG0, AccelFs_t(AccelFs::g16) | AccelOdr_t(AccelOdr::kHz1)));
+        PT_CALL(imu.updateRegister(Register::PWR_MGMT0, GyroMode_t(GyroMode::LowNoise) | AccelMode_t(AccelMode::LowNoise)));
+
+        while (true)
+        {
+            if (interrupt)
+            {
+                PT_CALL(imu.readRegister(Register::INT_STATUS, intStatus.value));
+                interrupt = false;
+            }
+
+            if (intStatus.any(IntStatus::FIFO_FULL_INT | IntStatus::FIFO_THS_INT))
+            {
+                if (PT_CALL(imu.readFifoData()))
+                {
+                    // Count packets in FIFO buffer and print contents of last packet
+                    uint16_t count = 0;
+                    modm::ixm42xxxdata::FifoPacket fifoPacket;
+                    for (const auto &packet : fifoData)
+                    {
+                        fifoPacket = packet;
+                        count++;
+                    }
+
+                    int16_t temp;
+                    modm::Vector<int32_t, 3> accel;
+                    modm::Vector<int32_t, 3> gyro;
+
+                    fifoPacket.getTemp(temp);
+                    fifoPacket.getAccel(accel);
+                    fifoPacket.getGyro(gyro);
+
+                    MODM_LOG_INFO.printf("Temp: %d\n", temp);
+                    MODM_LOG_INFO.printf("Accel: (%li, %li, %li)\n", accel.x, accel.y, accel.z);
+                    MODM_LOG_INFO.printf("Gyro: (%li, %li, %li)\n", gyro.x, gyro.y, gyro.z);
+                    MODM_LOG_INFO.printf("FIFO packet count: %d\n", count);
+                }
+                intStatus.reset(IntStatus::FIFO_FULL_INT | IntStatus::FIFO_THS_INT);
+            }
+        }
+        PT_END();
+    }
+
+private:
+
+    /// Due to the non-deterministic nature of system operation, driver memory allocation should always be the largest size of 2080 bytes.
```

Comment:
Good point! This comment is taken from the fifo documentation of the device. I believe it is for use with the FIFO_FULL_INT, when 2080 bytes of memory can be expected. However, for my own use case I have three imus so having three 2080 byte arrays may be a bit too much, especially if there is a lot of application code. What I may do is to set the template argument default 2080?

Reply:
Yes, that's a good option.

Reply:
I tried to do this, but when creating an instance of FifoData, I would then have to write FifoData<> to take the default template argument. Otherwise I get the warning error: invalid use of template-name 'modm::ixm42xxxdata::FifoData' without an argument list. Should I just leave it as is?

Reply:
Yes.